mesonbuild/environment.py

```diff
@@ -222,7 +222,7 @@ def detect_cpu_family(compilers):
                 trial = 'ppc64'
         if 'powerpc64' in stdo:
             trial = 'ppc64'
-    elif trial in ('amd64', 'x64'):
+    elif trial in ('amd64', 'x64', 'i86pc'):
```

Comment:
I don't think this is right, IIRC solaris return `i86pc` for x86 and x86_64. I think you need to do another test to determine which it is. @alanc has a branch with the correct code IIRC. I also have a copy of that branch I think.

Reply:
I looked into the respective line of @alanc. To me it seems reasonable in a way since Solaris/OpenIndiana has dropped 32bit support as far as I know.

Reply:
It looks like just 32bit kernels are deprecated, but 32bit userlands are still supported, which means we still need 32bit support as someone might need to compile for 32 bit (say, 32 bit mesa to run some legacy cad app)

Reply:
Correct - in Solaris 11 and later, the kernel is always 64-bit, but both 32-bit & 64-bit libraries & applications can be run.

Reply:
Well, "uname -a" shows "SunOS openindiana 5.11 illumos-9850293dd7 i86pc i386 i86pc".
The tests where that statement is in the function detect_cpu_family and that is i86pc for the "intel"-version of Solaris.

Talking about the cpu/cpu-family it is then x86_64, of course one could also compile i686. Having that in mind I do not understand why detect_cpu_family should differ from the real amd64 cpu which is the only one Solaris runs on intel systems... The topic you mention seems more  related to some kind of "cross-compiling".

The code I used was mainly the one from @alac. and combined with the correct cpu family, well the one Solaris offers, it seems correct to me.

How would you do the test for x86 or x64 here then?

Reply:
I got some info from the tests which seem reasonable and based on those lines:
System information.
Architecture: ('64bit', 'ELF')
Machine: i86pc
Platform: SunOS
Processor: i386
System: SunOS

Reply:
You're correct. It should always return x86_64 for the build machine.